sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionSerializable.java

```diff
@@ -32,78 +33,11 @@
-        ReflectiveInvoker xmlWriterWriteXmlSerializable = null;
-        ReflectiveInvoker xmlWriterFlush = null;
-        boolean xmlSerializableSupported = false;
-        try {
-            xmlSerializable = Class.forName("com.azure.xml.XmlSerializable");
-            xmlReader = Class.forName("com.azure.xml.XmlReader");
-            Class<?> xmlWriter = Class.forName("com.azure.xml.XmlWriter");
-
-            try {
-                Class<?> xmlProviders = Class.forName("com.azure.xml.XmlProviders");
-
-                xmlReaderCreator = ReflectionUtils.getMethodInvoker(xmlProviders,
-                    xmlProviders.getDeclaredMethod("createReader", byte[].class));
-
-                xmlWriterCreator = ReflectionUtils.getMethodInvoker(xmlProviders,
-                    xmlProviders.getDeclaredMethod("createWriter", OutputStream.class));
-            } catch (LinkageError | ReflectiveOperationException ex) {
-                xmlReaderCreator = ReflectionUtils.getMethodInvoker(xmlReader,
-                    xmlReader.getDeclaredMethod("fromBytes", byte[].class));
-                xmlWriterCreator = ReflectionUtils.getMethodInvoker(xmlWriter,
-                    xmlWriter.getDeclaredMethod("toStream", OutputStream.class));
-            }
-
-            xmlWriterWriteStartDocument
-                = ReflectionUtils.getMethodInvoker(xmlWriter, xmlWriter.getDeclaredMethod("writeStartDocument"));
-
-            xmlWriterWriteXmlSerializable
-                = ReflectionUtils.getMethodInvoker(xmlWriter, xmlWriter.getDeclaredMethod("writeXml", xmlSerializable));
-
-            xmlWriterFlush = ReflectionUtils.getMethodInvoker(xmlWriter, xmlWriter.getDeclaredMethod("flush"));
-
-            xmlSerializableSupported = true;
-        } catch (Throwable e) {
-            if (e instanceof LinkageError || e instanceof Exception) {
-                LOGGER.log(LogLevel.VERBOSE, () -> "XmlSerializable serialization and deserialization isn't supported. "
-                    + "If it is required add a dependency of 'com.azure:azure-xml', or another dependencies which "
-                    + "include 'com.azure:azure-xml' as a transitive dependency. If your application runs as expected "
-                    + "this informational message can be ignored.", e);
-            } else {
-                throw (Error) e;
-            }
-        }
-
-        XML_SERIALIZABLE = xmlSerializable;
-        XML_READER = xmlReader;
-        XML_READER_CREATOR = xmlReaderCreator;
-        XML_WRITER_CREATOR = xmlWriterCreator;
-        XML_WRITER_WRITE_XML_START_DOCUMENT = xmlWriterWriteStartDocument;
-        XML_WRITER_WRITE_XML_SERIALIZABLE = xmlWriterWriteXmlSerializable;
-        XML_WRITER_FLUSH = xmlWriterFlush;
-        XML_SERIALIZABLE_SUPPORTED = xmlSerializableSupported;
-        FROM_XML_CACHE = XML_SERIALIZABLE_SUPPORTED ? new ConcurrentHashMap<>() : null;
+        FROM_XML_CACHE = new ConcurrentHashMap<>();
```

Comment:
Should we have a limit on the size of this cache?

Reply:
We do have a limit: https://github.com/Azure/azure-sdk-for-java/pull/39524/files/b31df7782f3afb2650483028ffe689b5a2f4771e#diff-81c191c5c4f7986f23f1c2a022f92d0babdbca2bdd3b2f623493fbccccc6dd74R289